src/Core/Domain/ValueObject/Reduction.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\ValueObject;
+
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+
+/**
+ * Provides valid reduction values
+ */
+class Reduction
+{
+    /**
+     * For reducing concrete amount of money from price
+     */
+    const TYPE_AMOUNT = 'amount';
+
+    /**
+     * For reducing certain percentage calculated from price
+     */
+    const TYPE_PERCENTAGE = 'percentage';
+
+    /**
+     * Maximum allowed value for percentage type reduction
+     */
+    const MAX_ALLOWED_PERCENTAGE = 100;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var Number
+     */
+    private $value;
+
+    /**
+     * @param string $type
+     * @param float $value
+     *
+     * @throws DomainConstraintException
+     */
+    public function __construct(string $type, float $value)
+    {
+        $this->assertIsAllowedType($type);
+        $this->assertIsValidValue($type, $value);
+        $this->type = $type;
+        $this->value = new Number((string) $value);
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return Number
+     */
+    public function getValue(): Number
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param string $type
+     *
+     * @throws DomainConstraintException
+     */
+    private function assertIsAllowedType(string $type)
+    {
+        $allowedTypes = [
```

Comment:
Same comment, you already use this here: https://github.com/PrestaShop/PrestaShop/pull/13716/files#diff-e69e9e26daa40214fc8894cd1f203501R87